### PR TITLE
ENH Add generic types

### PR DESCRIPTION
**Pull request by GuySartorelli**
Requires https://github.com/silverstripe/silverstripe-framework/pull/11108 and the corresponding silverstripe/cms PR for these generics to take effect.

## Issue
- https://github.com/silverstripe/silverstripe-framework/issues/11066

### CODE REVIEW

File: src/Extensions/CwpCommentingExtension.php [hunk 1]
```diff
@@ -2,14 +2,15 @@
...
+use SilverStripe\Comments\Controllers\CommentingController;
 use SilverStripe\Control\Email\Email;
 use SilverStripe\Core\Extension;
 use SilverStripe\Forms\Form;
 
 /**
  * Customises the comment form to conform to government usability standards
  *
- * {@see CommentingController}
+ * @extends DataExtension<CommentingController>
```
Comment (emteknetnz) [inline src/Extensions/CwpCommentingExtension.php]:
```suggestion
 * @extends Extension<CommentingController>
```

File: src/PageTypes/EventHolderController.php [hunk 2]
```diff
@@ -16,6 +16,8 @@
  *
  * When the user clicks on a month, pagination will be reset, but tags retained. Pagination retains all other
  * parameters.
+ *
+ * @extends DatedUpdateHolderController<NewsHolder>
```
Comment (emteknetnz) [inline src/PageTypes/EventHolderController.php]:
```suggestion
 * @extends DatedUpdateHolderController<EventHolder>
```
? I'm not sure if this is correct or not

**Review by emteknetnz — CHANGES_REQUESTED** [review]

File: src/Extensions/CwpCommentingExtension.php (hunk 1 above)
Comment (ChloeSartorelli) [reply src/Extensions/CwpCommentingExtension.php]: Done

File: src/PageTypes/EventHolderController.php (hunk 2 above)
Comment (ChloeSartorelli) [reply src/PageTypes/EventHolderController.php]: Oops! Yup good catch. That was a copy paste error.
Fixed.

**Review by emteknetnz — APPROVED** [review]

**Pull request merged**